provider/github/diff.go

```diff
@@ -45,20 +48,41 @@ type posRange struct {
 	RelStart, RelEnd int
 }
 
+type parsedFile struct {
+	ranges     []*posRange
+	linesAdded map[int]bool
+}
+
 func newDiffLines(cc *github.CommitsComparison) *diffLines {
 	return &diffLines{
 		cc:     cc,
-		parsed: make(map[string][]*posRange, len(cc.Files)),
+		parsed: make(map[string]*parsedFile, len(cc.Files)),
 	}
 }
 
-func (d *diffLines) ConvertLine(file string, line int) (int, error) {
-	ranges, err := d.ranges(file)
+// ConvertLine takes a line number on the original file, and returns the
+// corresponding line number in the patch diff. It will return ErrLineOutOfDiff
+// if the line falls outside of the diff (changed lines plus context).
+// With strict set to true, ErrLineNotAddition will be returned for lines
+// that are not an addition (+ lines in the diff).
+func (d *diffLines) ConvertLine(file string, line int, strict bool) (int, error) {
+	parsedFile, err := d.parseFile(file)
+	if err != nil {
+		return 0, err
+	}
+
+	diffLine, err := d.convertLine(parsedFile.ranges, line)
 	if err != nil {
 		return 0, err
 	}
 
-	return d.convertLine(ranges, line)
+	if strict {
+		if !parsedFile.linesAdded[diffLine] {
+			return 0, ErrLineNotAddition.New()
```

Comment:
Is there a chance we could pass actual file name and line number to the error? 
I find having such information invaluable when debugging something later

Reply:
We log that debug info, but in poster.go: https://github.com/src-d/lookout/pull/210/files#diff-874f7f13213445f79bf1b3482a66e541R181